feat: Implement contextual prompt suggestions via magnifying glass icon

ngoiyaeric opened the pull request:
### **User description**
This commit introduces functionality to the magnifying glass icon in the mobile icons bar.

When you click the magnifying glass icon:
- If the chat input field contains text, the system calls an AI service to generate a more verbose and contextual version of your prompt, updating the input field with the suggestion.
- If the chat input field is empty, the system calls an AI service to generate a new set of diverse, contextual example prompts, updating the suggestions displayed in the empty screen view.

Changes include:
- Added `getEnhancedPrompt` and `getNewExamplePrompts` server actions in `lib/actions/suggestions.ts` to interact with the AI model.
- Modified `components/mobile-icons-bar.tsx` to add an `onClick` handler to the existing Search icon, which calls the new suggestion actions.
- Updated `components/chat.tsx` to manage state for `input` and `exampleMessages` and pass them and their setters to `MobileIconsBar`, `ChatPanel`, and `EmptyScreen`.
- Modified `components/empty-screen.tsx` to dynamically display `exampleMessages` received via props.
- State management ensures that UI components are updated reactively based on AI suggestions.


___

### **PR Type**
Enhancement


___

### **Description**
- Add AI-powered prompt enhancement and example prompt generation.
  - Magnifying glass icon now enhances input or generates new examples.
  - Integrates `getEnhancedPrompt` and `getNewExamplePrompts` server actions.

- Refactor state management for prompt input and example prompts.
  - Passes state and setters through chat, icons bar, and empty screen.
  - Example prompts now dynamically update based on AI suggestions.

- Update UI components to support dynamic prompt suggestions.
  - `EmptyScreen` displays example prompts via props.
  - `MobileIconsBar` handles prompt enhancement and example refresh.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>suggestions.ts</strong><dd><code>Add AI server actions for prompt enhancement and example generation</code></dd></summary>
<hr>

lib/actions/suggestions.ts

<li>Add <code>getEnhancedPrompt</code> for AI-powered prompt rewriting.<br> <li> Add <code>getNewExamplePrompts</code> for generating new example prompts.<br> <li> Define Zod schemas for AI response validation.<br> <li> Handles errors gracefully, returning fallbacks.


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/184/files#diff-4a36cf0d6ecd288e9b53224becd3ebab2ea7f8132826ab9ce1a51511c470c679">+70/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>chat.tsx</strong><dd><code>Refactor chat state and pass prompt/example state to children</code></dd></summary>
<hr>

components/chat.tsx

<li>Add state for <code>input</code> and <code>exampleMessages</code>.<br> <li> Initialize example prompts with icons and headings.<br> <li> Pass input and example prompt state/setters to <code>MobileIconsBar</code>, <br><code>ChatPanel</code>, and <code>EmptyScreen</code>.<br> <li> Ensure <code>EmptyScreen</code> receives dynamic example prompts.


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/184/files#diff-015689ed51facf46aa073450a0660cfa8b56e37b6b94248a56ec66550007f31b">+18/-3</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>empty-screen.tsx</strong><dd><code>Refactor to accept dynamic example prompts via props</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

components/empty-screen.tsx

<li>Refactor to receive <code>exampleMessages</code> via props.<br> <li> Remove static example prompts from component.<br> <li> Support optional icons for each example prompt.<br> <li> Update button rendering to handle missing icons.


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/184/files#diff-b72e49874aacc393d0ca5153479d2effea7b981562b86157c6473a941ff73df4">+16/-30</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>mobile-icons-bar.tsx</strong><dd><code>Add magnifying glass AI prompt enhancement and example refresh</code></dd></summary>
<hr>

components/mobile-icons-bar.tsx

<li>Add props for current input and example prompt state/setters.<br> <li> Implement magnifying glass click to enhance input or refresh examples <br>via AI.<br> <li> Integrate server actions for prompt enhancement and example <br>generation.<br> <li> Update button handler for prompt suggestion logic.


</details>


  </td>
  <td><a href="https://github.com/QueueLab/QCX/pull/184/files#diff-9f258d33f15a037a252e9debe448117217cb455ef07d7e024426ab00e7f90851">+33/-2</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

google-labs-jules[bot] committed:
feat: Implement contextual prompt suggestions via magnifying glass icon

This commit introduces functionality to the magnifying glass icon in the mobile icons bar.

When you click the magnifying glass icon:
- If the chat input field contains text, the system calls an AI service to generate a more verbose and contextual version of your prompt, updating the input field with the suggestion.
- If the chat input field is empty, the system calls an AI service to generate a new set of diverse, contextual example prompts, updating the suggestions displayed in the empty screen view.

Changes include:
- Added `getEnhancedPrompt` and `getNewExamplePrompts` server actions in `lib/actions/suggestions.ts` to interact with the AI model.
- Modified `components/mobile-icons-bar.tsx` to add an `onClick` handler to the existing Search icon, which calls the new suggestion actions.
- Updated `components/chat.tsx` to manage state for `input` and `exampleMessages` and pass them and their setters to `MobileIconsBar`, `ChatPanel`, and `EmptyScreen`.
- Modified `components/empty-screen.tsx` to dynamically display `exampleMessages` received via props.
- State management ensures that UI components are updated reactively based on AI suggestions.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/QueueLab/QCX?pullRequest=184) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/QueueLab/QCX?pullRequest=184) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/QueueLab/QCX?pullRequest=184) it.</sub>

google-labs-jules[bot] committed:
fix: Resolve TypeScript error to ensure successful build

I corrected a TypeScript type mismatch in `components/chat.tsx` related to the `exampleMessages` state. The `icon` property's optionality was causing a conflict with the `MobileIconsBar` component's expected props.

I introduced an `ExampleMessage` interface with an optional `icon` property and applied this type explicitly to the `exampleMessages` state and its initial value. This resolves the type conflict and allows the project to build successfully.

ngoiyaeric commented:
@coderabbitai full review
